aten/src/ATen/native/Attn.cpp

```diff
@@ -0,0 +1,33 @@
+#include <tuple>
+
+#include <ATen/Tensor.h>
+
+namespace at { namespace native {
+
+namespace {
+
+auto is_matrix(Tensor const& t) -> bool {
+  return t.sizes().size() == 2;
```

Comment:
nit: you also have `.ndim()` here

Reply:
looks like i need .dim() in c++, ndim is only an alias available in python.

I find functions like this as members of TensorImpl?

Reply:
Yes